processor/deltatocumulativeprocessor/internal/data/add.go

```diff
@@ -23,7 +37,101 @@ func (dp Histogram) Add(in Histogram) Histogram {
 	panic("todo")
 }
 
-// nolint
 func (dp ExpHistogram) Add(in ExpHistogram) ExpHistogram {
-	panic("todo")
+	switch {
+	case dp.Timestamp() >= in.Timestamp():
+		fail("out of order")
+	case dp.Scale() != in.Scale():
+		fail("scale changed")
+	case dp.ZeroCount() != in.ZeroCount():
+		fail("zero count changed")
+	}
+
+	aggregate := func(dpBuckets, inBuckets pmetric.ExponentialHistogramDataPointBuckets) {
+		var (
+			dp   = Buckets{data: dpBuckets.BucketCounts(), offset: int(dpBuckets.Offset())}
+			in   = Buckets{data: inBuckets.BucketCounts(), offset: int(inBuckets.Offset())}
+			aggr = Buckets{data: pcommon.NewUInt64Slice()}
+		)
+		aggr.offset = int(min(dpBuckets.Offset(), inBuckets.Offset()))
```

Comment:
Hey. I was wondering if we take into account the negative offset  here for the positive buckets
```
Use case:

 // dp.offset = -2
 // dp.bucket_counts = [1,2]

 // in.offset = -5
// in.bucket_counts = [1,2,3]

If I understand the aggregation correctly the expected aggr is:
// aggr.offset = -5 (which is should be calculated correctly from this line)
// aggr.bucket_counts = [1,2,3,1,2] (which doesn't seem to be calculated correctly from the for loop)
```
I am new here, so feel free to correct if its already discussed or its supposed to be supported in future PRs. Thanks in advance for your comments

Reply:
that's a great catch! I missed that the offset can be negative at all ._.